src/main/java/com/paneedah/weaponlib/compatibility/Compatibility1_12_2.java

```diff
@@ -104,6 +104,7 @@ public class Compatibility1_12_2 implements Compatibility {
 
     private static final List<Block> blocksToCheck = Arrays.asList(
             Blocks.AIR,
+            Blocks.BARRIER,
```

Comment:
No barriers should not be penetrable, it's an easy way for server admins to protect things and create bullet blocker

Reply:
If the player has enabled PenetratesGlass and I add BARRIER, is this feasible?

Reply:
No, barrier blocks are meant to stop everything even in vanilla they are an all mighty stop everything tool.

Reply:
How about we just add an entirely separate config value for this?

`BulletPenetratesBarrier=false` by default.

Then if the player chooses to enable that, then add it to the list.

Reply:
> In some cases, a barrier may not be suitable as a block to block bullets, as people may only hope it can block players

That would be a wrong assumption, vanilla behavior is that it block all entities including projectiles like arrows, fire charge, ender pearl, etc...

Maybe in the future will I or somebody else create a mod for customizable barriers (that could be very cool and useful).

Reply:
Regardless, I actually believe this could be really useful for a personal project I've been working on previously, so if he can get it added to the config and make it configurable - hell yes I want it.

More configuration can never hurt. The more the better.

Reply:
> How about we just add an entirely separate config value for this?
> 
> `BulletPenetratesBarrier=false` by default.
> 
> Then if the player chooses to enable that, then add it to the list.

That would be an option (even though it might break stuff).

Because what you are doing is making the bullet not stop itself when in contact of a barrier block, but the barrier block will most likely still attempt to stop the bullet as it is an entity, like the player or any other mobs.

Reply:
So it might create wondering bullets, that just hovers in space statically just slowly creating lag

Reply:
This is why we test stuff before we push it! I believe it won't cause any issues but of course, please do test it first.

Reply:
> This is why we test stuff before we push it! I believe it won't cause any issues but of course, please do test it first.

Yeah, be sure to test before pushing.

But if the barrier block is correctly made it should just prevent the bullet from travelling further.

And before you try a hacky work around, no just teleporting the bullet in contact of a barrier block will not work, as it will also make it bypass any potential block behind that barrier block.